src/common/selectors/deploymentResultsSelector.ts

```diff
@@ -62,9 +62,11 @@ export async function selectOneDeploymentResultWithChangeBundleInstallsByAsyncJo
   asyncJobId: string
 ): Promise<CheckDeploymentResultWithChangeBundleInstalls | null> {
   const queryStr = `SELECT sf_devops__Check_Deploy__c, sf_devops__Deployment_Id__c, sf_devops__Check_Deploy_Status__r.sf_devops__Status__c,
-                    (SELECT sf_devops__Environment__c FROM sf_devops__Change_Bundle_Installs__r)
+                      (SELECT sf_devops__Environment__c FROM sf_devops__Change_Bundle_Installs__r)
                     FROM sf_devops__Deployment_Result__c  
-                    WHERE sf_devops__Check_Deploy_Status__c = '${asyncJobId}' AND sf_devops__Check_Deploy__c = TRUE`;
+                    WHERE sf_devops__Check_Deploy_Status__c = '${asyncJobId}'
+                      AND sf_devops__Check_Deploy__c = TRUE
+                      AND Id IN (SELECT sf_devops__Deployment_Result__c FROM sf_devops__Change_Bundle_Install__c)`;
```

Comment:
Just verifying a CBI exists is enough? or should we verify the status is completed.

Maybe that is handled elsewhere, and just needed to be sure CBI was present

Reply:
Yes, the status is validated in `abstractQuick.ts`. We don't filter the query by that so we get the current status and show a proper error message if needed.
For the case of missing CBIs we should show the same error message as if the deployment result were null so that's way added the filter to the query.